app/src/main/java/com/vero/tasky/auth/presentation/login/LoginViewModel.kt

```diff
@@ -54,23 +53,27 @@ class LoginViewModel @Inject constructor(
         }
     }
 
-    private fun logIn() {
-        EmailErrorHandler.validateEmail(
-            isEmailValid = state.isEmailValid,
-            onEmailIsValid = ::onEmailIsValid,
-            showError = ::showError
-        )
+    private fun validateUserInfo() {
+        if (state.isEmailValid) {
+            validatePassword()
+        } else {
+            showError(R.string.email_not_valid)
+        }
     }
 
-    private fun onEmailIsValid() {
-        PasswordErrorHandler.onPasswordValid(
-            passwordValidationResult = loginUseCases.validatePasswordUseCase(state.password),
-            onPasswordIsValid = ::onPasswordIsValid,
-            showError = ::showError
+    private fun validatePassword() {
+        val result = PasswordValidator.validatePassword(
+            passwordValidationResult = loginUseCases.validatePasswordUseCase(state.password)
```

Comment:
This could be separated into variables, to make it easier to read

val validation= loginusecase...

val result = PasswordValidator.validate(validation)